### ISSUE TITLE
`as.Date()` now throws an error (version 2.3.2) when querying a Teradata server.

### ISSUE THREAD

**Issue by SayfSaid** [tidyverse/dbplyr#1285]
I had a previously running code to query a Teradata server that is no longer running after updating (dbplyr, R, and odbc).
The issue seems to be related to how `as.Date()` is being translated into SQL. Using `show_query()`, in the prior versions it came as CAST("X" AS DATE), now SELECT DATE "X" which throws an error when trying to pull the data.

I know it is related to the update because the code still runs on a VM running older versions and querying the same server.
The updates are: R 4.2 to 4.3, dplyr 1.0.10 to 1.1.2, dbplyr 2.2.1 to 2.3.2, odbc 1.3.3 to 1.3.4, and DBI is the same 1.1.3.

Below are comparison of the two queries.

```r
   library (dplyr)
   library (DBI)
   library (odbc)
   connection <- dbConnect(odbc(), "Teradata")

   tbl(connection, "Teradata_table") %>% 
   head() %>% 
   select(POSIXct_POSIXt_var) %>% 
   mutate(date_var= as.Date(POSIXct_POSIXt_var)) %>% 
   show_query()
```
The old version brings 
```
<SQL>
   SELECT "POSIXct_POSIXt_var", CAST("POSIXct_POSIXt_var" AS DATE) AS "date_var"
   FROM (
     SELECT TOP 6 *
     FROM "Teradata_table"
   ) "q01"
```
Now results in: 

```
 <SQL>
   SELECT TOP 6 "POSIXct_POSIXt_var", DATE "POSIXct_POSIXt_var" AS "date_var"
   FROM "Teradata_table"

```

The latter throws an error when trying to pull the data as such: 

```r
   tbl(connection, "Teradata_table") %>% 
   head() %>% 
   select(POSIXct_POSIXt_var) %>% 
   mutate(date_var = as.Date(POSIXct_POSIXt_var)) %>% 
   as_tibble()
```
Results in: 

> (-3706)Syntax error: expected something between the word 'POSIXct_POSIXt_var' and the 'AS' keyword.

While I had a good suggestion for a walk around on my [SO post](https://stackoverflow.com/q/76282760/13695652) by using `sql('CAST ("var" AS DATE)')` to replace `as.Date()`,  I feel this might need to be addressed here to retain `as.Date()` functionality for Teradata use. A more experience [user](https://stackoverflow.com/users/1082435/wibeasley) on SO suggested that the issue might be related to [this commit](https://github.com/tidyverse/dbplyr/commit/83045c593a69c0d981bad0051cf63be24e923c49#diff-c4f086980e5692d16e3337acd7bfefe3a9315135688133ee4f24ea2944ccba3bL113). I hope this helps.

**Comment by mgirlich** [comment]
@overmar This translations was added by you in #913. Would be great if you could look into this.

**Comment by But2ene** [comment]
The problem comes from two conflicting use cases.

# Litterals
With litterals such as "2020-01-01", what was implemented in #913 is correct. It is a good practice to prefix date litterals with the keyword "DATE". For example,
```r
as.Date("2020-01-01")
```
should be translated to (with single quotes !)
```sql
DATE '2020-01-01'
```

# Identifiers
With identifiers / variable names / vectors of length greater than 1, we have
```r
as.Date(MY_TIMESTAMP_VAR)
```
should be translated to (with double quotes)
```sql
CAST("MY_TIMESTAMP_VAR" AS DATE)
```

I'm not sure how to implement something that can distinguish both cases. By default, I would expect `as.Date` and `lubridate::as_date` to be both translated as `CAST AS DATE`, which was the previous default. In my code, I manually specify the `DATE '2020-01-01'` when needed.

**Comment by mgirlich** [comment]
Fixed by [8e4feba](https://github.com/tidyverse/dbplyr/commit/8e4feba4f27b2e88512b37415606f3349d58113c).

**Comment by mgirlich** [comment]
Thanks @But2ene for the detailed description.
